rotkehlchen/chain/ethereum/modules/nfts.py

```diff
@@ -312,6 +314,19 @@ def delete_price_for_nft(self, asset: Asset) -> bool:
 
         return True
 
+    def _filter_ignored_nfts(self, nfts_data: Dict[ChecksumEvmAddress, List[NFT]]) -> Dict[ChecksumEvmAddress, List[NFT]]:  # noqa: E501
+        """Remove ignored NFTs from NFTs data."""
+        with self.db.conn.read_ctx() as cursor:
+            ignored_nfts = set(self.db.get_ignored_assets(cursor=cursor, only_nfts=True))
+
+        nfts_data_copy = nfts_data.copy()
+        for nfts in nfts_data_copy.values():
+            for nft in nfts:
+                if nft.token_identifier in ignored_nfts:
+                    nfts.remove(nft)
```

Comment:
what are you doing here? Why are you modifying the values list while you are iterating it? Why are you making a shallow copy of the nfts_data dict? I am really confused. There is no comments and the code is rather hard to read and non-performant. And you are using the `remove` function of a list. Do you realize how bad for perormance this is?

Why not instead do something close to this?

```suggestion
        for address, nfts in nfts_data.items():
           nfts_data[address]  =  [x for x in nfts if x.identifier not in ignored_nfts]
```

Simpler, easier to read and understand and much much more performant unless I misunderstand some constraint you had.

Reply:
okay.

Reply:
Answering your questions:

> Why are you making a shallow copy of the nfts_data dict? I am really confused
To avoid directly mutating the cached data. That is the reason for the shallow copy.

Reply:
> Answering your questions:
> 
> > Why are you making a shallow copy of the nfts_data dict? I am really confused
> > To avoid directly mutating the cached data. That is the reason for the shallow copy.

but with the approch I suggested this is not needed, right?